eventstore/memory/memory.go

```diff
@@ -104,16 +80,24 @@ func (e *Memory) Get(ctx context.Context, id string, aggregateType string, after
 	return &iterator{events: events}, nil
 }
 
-// GlobalEvents will return count events in order globally from the start posistion
-func (e *Memory) GlobalEvents(start, count uint64) ([]core.Event, error) {
+// Close does nothing
+func (e *Memory) Close() {}
+
+// aggregateKey generate a aggregate key to store events against from aggregateType and aggregateID
+func aggregateKey(aggregateType, aggregateID string) string {
+	return aggregateType + "_" + aggregateID
+}
+
+// globalEvents will return count events in order globally from the start posistion
+func (e *Memory) globalEvents(start core.Version, count uint64) ([]core.Event, error) {
 	var events []core.Event
```

Comment:
Given that we know that maximum count, or capacity, of the `events` array, would it make sense to create an array using the make function instead?

```suggestion
	events := make([]core.Event, count)
```

But that is based on the assumption that there will be atleast `count` events.
In this case we would have to return a slice:
`return events[:count]`

Reply:
```suggestion
	events := make([]core.Event, 0, count)
```
Setting the capacity is maybe the best middle ground?

Reply:
Yes, that's probably what I meant!